### ISSUE TITLE
premake does not build on macOS 10.14 (Mojave)

### ISSUE THREAD

**Issue by fxcoudert** [premake/premake-core#1154]
premake does not build on macOS 10.14 (Mojave) because it contains numerous references to `-mmacosx-version-min=10.4` as compiler flag… which is now unsupported. The error message is:

```
$ make -C build/gmake.macosx
==== Building Premake4 (release) ====
Linking Premake4
ld: library not found for -lgcc_s.10.4
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [../../bin/release/premake4] Error 1
make: *** [Premake4] Error 2
```

**Comment by fxcoudert** [comment]
Showed up during Homebrew testing: https://github.com/Homebrew/homebrew-core/issues/28817

**Comment by samsinsane** [comment]
I don't fully understand what the minimum version does on macOS, will letting it target the current version result in the binaries not running on earlier versions of macOS? Or does the minimum version just us from using certain APIs?

**Comment by fxcoudert** [comment]
Yes, binaries built without will only run on later macOS versions.

**Comment by fxcoudert** [comment]
Most software has no `--macosx-version-min`, so it does like this:
- by default it builds for the current architecture (and will always run on later OS versions)
- if the user (or distro) wants to build a version that can run on older OS that it is built on, they can specify `--macosx-version-min` themselves through the environment variable (`export MACOSX_DEPLOYMENT_TARGET=10.9`)

Right now, build is simply broken on macOS 10.13 and 10.14, which is 90% of macOS users.

**Comment by tdesveauxPKFX** [comment]
I think we should remove the osx min version from our Premake script and add the define to our release process. This way, you can target whichever osx version you want when you clone and our prebuilt binaries works the same.

From the log posted in the first message, the issue seems to be with Premake4. Also, do we know who own the package? Or does it clone the repo then build?

**Comment by na-Itms** [comment]
Hello, commit https://github.com/premake/premake-core/commit/5f57b5d62a4518f391cf66e1cfb70e4a92353178 removed the hardcoded value, but left it in [premake4.lua](https://github.com/premake/premake-core/blob/master/premake4.lua), and did not change the packaging scripts. Was it a thoughtful decision regarding this issue? The change doesn't look like matching the ideal plan laid out in comments above.

We are currently hitting the bug because we are still on alpha 10, and we would like to make sure the issue is properly fixed before going through a premake upgrade.

Thank you very much!

**Comment by na-Itms** [comment]
> Regarding the "packaging scripts", which ones are you referring to? I can't find any reference to `-mmacos-version-min` in the entire repo except for the `premake4.lua` file.

Sorry, I was referring to this phrase by @tdesveauxPKFX:

> I think we should remove the osx min version from our Premake script and add the define to our release process. 

Basically I was pointing out that the min version was removed but the commit didn't add any define in any place. I assumed your release process was carried out using packaging scripts in which defines could be added.

> Upgrading beyond alpha10 shouldn't introduce any problems to your projects, but in any case it is resolved in alpha13. If you have any issues with alpha13, please feel free to report them.

We are currently in the latest stages of releasing a new version, so we will only make minimal modifications and apply commit 5f57b5d to our alpha10 copy. We will then upgrade premake ASAP. I will be sure to contact you in case we need some help 👍
